### ISSUE TITLE
Error fetching sheet

### ISSUE THREAD

**Issue by ryan-williams** [samcday/node-google-spreadsheets#35]
```sh
$ cat gsheet.js
#!/usr/bin/env node

var GoogleSpreadsheets = require("google-spreadsheets");

var key = process.argv[2];

console.log("sheet:", key);

GoogleSpreadsheets(
      {
        key: key
      },
      function(err, sheet) {
        console.log(err, sheet);
      }
);
$ gsheet.js 1Bb-SLy4E-alUUSJ68Qb5nNmZG6HsPbG2j6Ye_QZO77Y
sheet: 1Bb-SLy4E-alUUSJ68Qb5nNmZG6HsPbG2j6Ye_QZO77Y
[TypeError: Cannot read property 'title' of undefined] null
```

Trying to fetch [this sheet](https://docs.google.com/spreadsheets/d/1Bb-SLy4E-alUUSJ68Qb5nNmZG6HsPbG2j6Ye_QZO77Y/edit#gid=0).

**Comment by samcday** [comment]
I think this is a dupe of #34?

**Comment by ryan-williams** [comment]
Pretty similar: the sheet didn't have the right permissions, as you noted.

Can I rephrase this as: "Improve error message when fetching sheet with incorrect permissions"?

I filed this separately from #34 because it goes via a different API and gives a different, also unhelpful, error message to what I was hitting in #34.

**Comment by samcday** [comment]
This issue is now fixed in `0.5.0` of the lib. If you try to access a non-public sheet without auth you'll get the error `No access to that spreadsheet, check your auth.`.
